scripts/deploy-tarball.js

```diff
@@ -0,0 +1,149 @@
+const path = require('path')
+const execa = require('execa')
+const fs = require('fs/promises')
+
+const cwd = process.cwd()
+
+async function main() {
+  const deployDir = path.join(cwd, 'files')
+  const publicDir = path.join(deployDir, 'public')
+  await fs.mkdir(publicDir, { recursive: true })
+  await fs.writeFile(
+    path.join(deployDir, 'package.json'),
+    JSON.stringify({
+      name: 'files',
+      dependencies: {},
+      scripts: {
+        build: 'node inject-deploy-url.js',
+      },
+    })
+  )
+  await fs.copyFile(
+    path.join(cwd, 'scripts/inject-deploy-url.js'),
+    path.join(deployDir, 'inject-deploy-url.js')
+  )
+
+  let nativePackagesDir = path.join(cwd, 'packages/next-swc/crates/napi/npm')
+  let platforms = (await fs.readdir(nativePackagesDir)).filter(
+    (name) => !name.startsWith('.')
+  )
+
+  const optionalDeps = {}
+
+  await Promise.all(
+    platforms.map(async (platform) => {
+      let version = JSON.parse(
+        await fs.readFile(path.join(cwd, 'lerna.json'))
+      ).version
```

Comment:
Line 35-37 should be hoisted out of the platforms.map() loop since it only needs to happen once.